src/models/user.ts

```diff
@@ -1,49 +1,55 @@
-import mongoose, { Schema } from "mongoose";
+import mongoose, { model, Schema } from 'mongoose'
+import bcrypt from "bcryptjs"
 
-const User = mongoose.model(
-  "User",
-  new Schema({
+const userSchema = new Schema({
     email: {
-      type: String,
-      required: true,
-      unique: true,
+        type: String,
+        required: true,
+        unique: true,
     },
     password: {
-      type: String,
-      required: true,
+        type: String,
+        required: true,
     },
     role: {
-      type: String,
-      default: "user",
+        type: String,
+        default: 'user',
     },
     profile: {
-      type: mongoose.Types.ObjectId,
-      ref: "Profile",
+        type: mongoose.Types.ObjectId,
+        ref: 'Profile',
     },
-  })
-);
+})
+
+userSchema.methods.checkPass = async function (password:string) {
+    const pass = await bcrypt.compare(password, this.password)
+    return pass
+}
+
+
+ const User = model("User", userSchema)
 
 const Profile = mongoose.model(
-  "Profile",
-  new Schema({
-    firstName: {
-      type: String,
-      required: true,
-    },
-    lastName: {
-      type: String,
-      required: true,
-    },
-    address: {
-      type: String,
-    },
-    user: {
-      type: mongoose.Types.ObjectId,
-      ref: "User",
-      required: true,
-      unique: true,
-    },
-  })
-);
+    'Profile',
+    new Schema({
+        firstName: {
+            type: String,
+            required: true,
+        },
+        lastName: {
+            type: String,
+            required: true,
+        },
+        address: {
+            type: String,
```

Comment:
I see changes related to indentation. There's no team-wide linting config?

Reply:
well resolved